### PR TITLE
Add ContractHash type to ensure reverse hex serialization

### PR DESCRIPTION
**Pull request by stevenroose**
Which is what Elements Core expects.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,6 +17,10 @@ documentation = "https://docs.rs/elements/"
 
 [dependencies]
 bitcoin = "0.23"
+# While this dependency is included in bitcoin, we need this to use the macros.
+# We should probably try keep this one in sync with the bitcoin version,
+# to avoid requiring two version of bitcoin_hashes.
+bitcoin_hashes = "0.7.6"
```
Comment (apoelstra) [inline Cargo.toml:23]: Looks like rust-bitcoin is using >0.7.3 ... I think this is okay. Technically it means that somebody could pin their version at 0.7.3 and cargo wouldn't complain, but this seems like a pretty unlikely scenario.

Comment (apoelstra) [reply Cargo.toml:23]: Ah @stevenroose says elsewhere that we need this to access the macros. Can you update the comment to reflect this?

Comment (stevenroose) [reply Cargo.toml:23]: The first line of the comment says we need this to access the macros :)

If someone pinned their version to 0.7.3, he wouldn't be able to build elements.

**Comment by stevenroose** [comment]
Fixed the unit test and rebased.

**Comment by shesek** [comment]
Nice, that's pretty useful, thanks! I used this for esplora/electrs and for the liquid asset registry, works for me as well.

**Pull request merged**